### PR TITLE
Fix #1196 - Gensim error when loading FastText

### LINKED ISSUE #1196: Gensim error when loading FastText
Hello,

I downloaded fasttext [English wikipedia](https://s3-us-west-1.amazonaws.com/fasttext-vectors/wiki.en.zip)

I simply executed `model = FastText.load_fasttext_format('/Users/smidj/Repositories/snippetbot/data/fasttext/wiki.en/wiki.en')`

I got an error:
```
OSError                                   Traceback (most recent call last)
<ipython-input-3-83dcc56dad43> in <module>()
----> 1 model = FastText.load_fasttext_format('/Users/smidj/Repositories/snippetbot/data/fasttext/wiki.en/wiki.en')

/Users/smidj/.virtualenvs/snippetbot/lib/python3.6/site-packages/gensim/models/wrappers/fasttext.py in load_fasttext_format(cls, model_file)
    236         model = cls()
    237         model.wv = cls.load_word2vec_format('%s.vec' % model_file)
--> 238         model.load_binary_data('%s.bin' % model_file)
    239         return model
    240 

/Users/smidj/.virtualenvs/snippetbot/lib/python3.6/site-packages/gensim/models/wrappers/fasttext.py in load_binary_data(self, model_binary_file)
    254             self.load_model_params(f)
    255             self.load_dict(f)
--> 256             self.load_vectors(f)
    257 
    258     def load_model_params(self, file_handle):

/Users/smidj/.virtualenvs/snippetbot/lib/python3.6/site-packages/gensim/models/wrappers/fasttext.py in load_vectors(self, file_handle)
    301 
    302         self.num_original_vectors = num_vectors
--> 303         self.wv.syn0_all = np.fromstring(file_handle.read(num_vectors * dim * float_size), dtype=dtype)
    304         self.wv.syn0_all = self.wv.syn0_all.reshape((num_vectors, dim))
    305         assert self.wv.syn0_all.shape == (self.bucket + len(self.wv.vocab), self.size), \

OSError: [Errno 22] Invalid argument
```

I use Mac OS Sierra, python 3.6 (in virtaulenv), and gensim version 1.0.1.

Thanks,
Jakub

### PR DESCRIPTION
**Pull request by jaksmid**
Gensim can load large fasttext model on Mac
Loading of binary fasttext models is faster
Fasttext vector size is correctly set
Vector_size is also saved when loading just the vector_file, which can be useful if you are not interested in training

### CODE REVIEW

**Comment by jaksmid** [comment]
I am getting cannot allocate memory in Travis build

**Comment by tmylk** [comment]
Travis tests re-ran after smart_open update

**Comment by tmylk** [comment]
What is the purpose of adding a new attribute `vector_size` to all KeyedVectors? isn't it always `syn0.shape[0]`? If necessary, maybe just add it to FastText class?

**Comment by jaksmid** [comment]
Thanks @tmylk for the comment. To my understanding, you would use FastText in the case you want to load both the vec and bin files of the fasttext. In case you just want to load the vectors you can use FastTextKeyedVectors. As you pointed out, you can use `syn0.shape[0]` which may not be that straightforward if you are not familiar with the gensim internals. So I thought it can be a good idea to add vector_size explicitly so people can figure out the right name of the attribute more easily.

By adding it to the FastText class you meant FastTextKeyedVectors class, right?
Thanks!

**Comment by jaksmid** [comment]
Furthermore, you cannot just use FastTextKeyedVectors without FastText initialization (which needs both vec and bin) as `__contains__` in `FastTextKeyedVectors` references `self.min_n` which is actually set in FastText `load_model_params`.

File: gensim/test/test_fasttext_wrapper.py [hunk 1]
```diff
@@ -169,10 +189,12 @@ def testMostSimilarCosmul(self):
         """Test most_similar_cosmul for in-vocab and out-of-vocab words"""
         # In vocab, sanity check
         self.assertEqual(len(self.test_model.most_similar_cosmul(positive=['the', 'and'], topn=5)), 5)
-        self.assertEqual(self.test_model.most_similar_cosmul('the'), self.test_model.most_similar_cosmul(positive=['the']))
+        self.assertEqual(self.test_model.most_similar_cosmul('the'),
```
Comment (tmylk) [inline gensim/test/test_fasttext_wrapper.py]: Please use hanging indent

File: gensim/test/test_fasttext_wrapper.py [hunk 2]
```diff
@@ -169,10 +189,12 @@ def testMostSimilarCosmul(self):
...
+                         self.test_model.most_similar_cosmul(positive=['the']))
         # Out of vocab check
         self.assertEqual(len(self.test_model.most_similar_cosmul(['night', 'nights'], topn=5)), 5)
-        self.assertEqual(self.test_model.most_similar_cosmul('nights'), self.test_model.most_similar_cosmul(positive=['nights']))
+        self.assertEqual(self.test_model.most_similar_cosmul('nights'),
```
Comment (tmylk) [inline gensim/test/test_fasttext_wrapper.py]: Hanging indent preferred

**Comment by jaksmid** [comment]
I incorporated the code review and merged with the current develop.

It seems like the build for python 2 stalled, could you rerun it @tmylk please?

**Comment by tmylk** [comment]
These tests are know to occasionally fail but it's the first time they fail constantly. Will disable them in the main branch soon.

**Comment by jaksmid** [comment]
Ok. Let me know if I can help with anything.

**Pull request merged**